gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -0,0 +1,330 @@
+use std::{collections::HashMap, path::PathBuf, sync::Arc};
+
+use cln_plugin::{options, Builder, Plugin};
+use cln_rpc::{model, ClnRpc};
+use fedimint_api::{task::TaskGroup, Amount};
+use fedimint_server::config::load_from_file;
+use ln_gateway::{
+    config::ClnRpcConfig,
+    gatewaylnrpc::{
+        gateway_lightning_server::{GatewayLightning, GatewayLightningServer},
+        GetPubKeyRequest, GetPubKeyResponse, PayInvoiceRequest, PayInvoiceResponse,
+        SubscribeInterceptHtlcsRequest, SubscribeInterceptHtlcsResponse,
+    },
+};
+use serde::{Deserialize, Deserializer, Serialize};
+use serde_json::json;
+use thiserror::Error;
+use tokio::{
+    io::{stdin, stdout},
+    sync::{mpsc, Mutex},
+};
+use tokio_stream::wrappers::ReceiverStream;
+use tonic::{transport::Server, Status};
+use tracing::error;
+
+// Note: One this binary is stable, we should be able to remove current 'ln_gateway'
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    let service = ClnRpcService::new()
+        .await
+        .expect("Failed to create cln rpc service");
+    let dir = service.dir.clone();
+
+    // Read configurations
+    let gw_cfg_path = dir.join("lnrpc.config");
```

Comment:
Can we just pass this in via command line or environment? Having so many configs is pretty annoying in deployment.

Reply:
Ack. Will do a makeover to allow reading configs from env variables. Same thing for other high level gatewayd configs. There is a lot of changes up in the air right now, so not prioritizing the config pattern change

Reply:
fixed by #1295